presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayMinMaxUtils.java

```diff
@@ -92,7 +94,10 @@ public static Double doubleArrayMinMax(MethodHandle compareMethodHandle, Type el
                     return null;
                 }
                 double value = elementType.getDouble(block, i);
-                if ((boolean) compareMethodHandle.invokeExact(value, selectedValue)) {
+                if (isNaN(value)) {
```

Comment:
nit: in fact this could be second `if, else` option, not the first. Thus, for array in decrementing order and w/o NaNs you do less comparisons